### ISSUE TITLE
No config saving? No Logging?

### ISSUE THREAD

**Issue by deconsult** [Nesos-ita/DynamicDnsUpdater#1]
Hi,
I try to update my DynDNS-Account at dedyn.io with DynamicDnsUpdater. I tried starting with standard user and admin privileges and as a startup task.
Here my questions and remarks:
It seems, that the log setting (for example App directory) in the GUI isn't saved? Or what do wrong?
The button View Log gives the error mesage that there is no log file. But the log file is written when starting an terminating the GUI.
The DynDNS-Updates should be logged?
A workaround for me:  what are the logging keywords and values in the dduSettings.txt file?

See my environment:
![dynamicdnsupdater errors](https://user-images.githubusercontent.com/34723168/34228655-b31c5b2a-e5d1-11e7-9ff5-697c6c3418b8.JPG)

Thanks for an answer
deconsult

**Comment by Nesos-ita** [comment]
Thanks for reporting, should be fixed with the latest version, download it and let me know.  

> It seems, that the log setting (for example App directory) in the GUI isn't saved? Or what do wrong?

Correct, it *was* not implemented (see below for reason).

> The button View Log gives the error mesage that there is no log file. But the log file is written when starting an terminating the GUI.

Was a bug, now fixed; the button always tried to open temp log.

> The DynDNS-Updates should be logged?

yes, in the log you should see an entry like this: `27/12/2017 13.21.40 - Updating DNS: OK` or an error in case there is one.
more info [in the FAQ](https://github.com/Nesos-ita/DynamicDnsUpdater/blob/master/Documentation/FAQ.md#what-is-the-log-option-and-what-does-it-log)

> A workaround for me: what are the logging keywords and values in the dduSettings.txt file?

as i said it was not implemented, because my idea was that logging in gui was useless because you have gui, and was more useful for background mode. But thanks to you i noticed that everything you do is saved except log setting and this is wrong, so now log setting is saved and used both in gui and background mode.
Also i have added a [file format specification](https://github.com/Nesos-ita/DynamicDnsUpdater/blob/master/Documentation/File%20format%20specifications.md) which explain all the config lines.

**Comment by Nesos-ita** [comment]
Since there is no answer i'll consider this solved, feel free to comment or open a new issue if you find something other.
Thanks again for the feedback!

**Comment by deconsult** [comment]
Sorry for the delay,
but loggging is now (almost) ok.
Little hint: if there was already a setup with the older version 1.0.1 with Windows start of DynamicDnsUpdater and logging, the task manager action line is
"C:\Program Files\DynamicDnsUpdater\DynamicDnsUpdater.exe" -bg -log
The -log switch prevents form logging. That isn't corrected by switching log mode in the gui. 
Solution: delete -log in the task manager action line 
Thanls for your work.

**Comment by Nesos-ita** [comment]
> with the older version 1.0.1 ... DynamicDnsUpdater.exe" -bg -log

this is known. i could force a change in the autorun value on program upgrade but it might require admin privilege so i have left it unchanged.
log options in gui are grayed out if you pass a log option by command line (but you can't notice this since on boot it runs in background...)

> The -log switch prevents form logging.

this shouldn't be the case.  
-log is a, now undocumented, alias for -LogTemp. i have not documented it because is now ambiguous and i plan to drop it in future versions. but for now if you are coming from the old version it should have log to temp forced on.  
this because as you noticed the _old_ autorun entry has the -log option, and command line log options override the saved one (is ignored).  
if you want to have the autorun entry to behave according to the saved options you can delete the -log option from the autorun entry as you said or more simply click "autorun no" and than re enable the autorun.  
new autorun entry is created without log option so it will behave the according to the saved setting (visible in gui).
i have added a note in the version history.  

TL;DR: click autorun no and re enable it so that the autorun entry is deleted and recreated. from now it will log on boot according to the saved setting.

Thanks for the feedback!
